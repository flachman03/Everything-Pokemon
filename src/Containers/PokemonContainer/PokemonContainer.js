import React from 'react'
import { PokemonCard } from '../../Components/PokemonCard/PokemonCard'
import './PokemonContainer.scss'
import { connect } from 'react-redux'
import { getMorePokemonThunk } from '../../Thunks/pokemonThunks'

export class PokemonContainer extends React.Component {
  constructor() {
    super();
    this.state = { 
      offset: 0,
    }
  }

  componentDidMount() {
    console.log(this.props)
  }
  
  handleNextPokemon = (offset) => {
    this.setState({offset: this.state.offset + offset})
    this.props.getMorePokemon(this.state.offset)
    console.log(this.state.offset)
  }

  handlePrevPokemon = (offset) => {
    this.setState({offset: this.state.offset - offset})
    this.props.getMorePokemon(this.state.offset)
  }

  render() {
    const { data } = this.props
    const   displayPokemon = data.map( pokemon => {
      return <PokemonCard pokemon={pokemon}/>
    })

    return (
      <section className="pokemon-section">
        <div className="pokemon__container">
          {displayPokemon}
        </div>
        <ul className="container__buttons">
          <button 
            className="poke-button"
            onClick={() => this.handlePrevPokemon(20)}
          >Previous</button>
          <button 
            className="poke-button"
            onClick={() => this.handleNextPokemon(20)}
          >Next</button>
        </ul>
      </section>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  getMorePokemon: (offset) => dispatch(getMorePokemonThunk(offset))
})

export default connect(null, mapDispatchToProps)(PokemonContainer);