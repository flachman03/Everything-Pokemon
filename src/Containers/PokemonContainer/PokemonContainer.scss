.pokemon__container {
  display: flex;
  flex-wrap: wrap;
}

.container__buttons {
  margin-top: 100px;
  display: flex;
  justify-content: center;
}

.poke-button {
  width: 100px;
  height: 30px;
  margin: 10px;
}
