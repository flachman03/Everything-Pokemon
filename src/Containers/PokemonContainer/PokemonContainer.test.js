import React from 'react'
import { shallow } from 'enzyme'
import { PokemonContainer, mapDispatchToProps } from './PokemonContainer'
// import { getMorePokemonThunk } from '../../Thunks/pokemonThunks'

describe('PokemonContainer', () => {
  let wrapper, mockData, getMorePokemon, getMorePokemonThunk;
  beforeEach(() => {
    getMorePokemon = jest.fn()
    getMorePokemonThunk = jest.fn()
    mockData = [{name: 'charmander'}]
    wrapper = shallow(<PokemonContainer data={mockData} getMorePokemon={getMorePokemon}/>)
    window.fetch = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        json: () => Promise.resolve(mockData)
      })
    })
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
  it('should have a default state', () => {
    const expected = {offset: 0}
    expect(wrapper.state()).toEqual(expected)
  })

  it('should have a method that adds offset to the state', async () => {
    const expected = {offset: 20}
    await wrapper.instance().handleNextPokemon(20)
    expect(wrapper.state()).toEqual(expected)
  })

  it('should call the function getMorePokemon with handling the next pokemon method', async () => {
    await wrapper.instance().handleNextPokemon(20)
    expect(getMorePokemon).toHaveBeenCalled()
    expect(getMorePokemon).toHaveBeenCalledWith(20)
  })

  it('should have a method that reduces offset from the state', async () => {
    await wrapper.instance().handleNextPokemon(20)
    expect(wrapper.state().offset).toEqual(20)
    await wrapper.instance().handlePrevPokemon(20)
    expect(wrapper.state().offset).toEqual(0)
  })

  it('should call the getMorePokemon function when handleing previous pokemon', async () => {
    await wrapper.instance().handlePrevPokemon(20)
    expect(getMorePokemon).toHaveBeenCalled()
  })

  // it('should mapDispatchToProps with the getMorePokemonThunk', () => {
  //   const mockDispatch = jest.fn()
  //   const offset = 20;
  //   const expected = { 'getMorePokemon': getMorePokemonThunk()}
  //   const result = mapDispatchToProps(mockDispatch)
  //   expect(result).toHaveBeenCalledWith(expected)
  // })
})