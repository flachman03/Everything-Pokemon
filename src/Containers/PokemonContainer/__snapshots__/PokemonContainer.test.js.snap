// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonContainer should match the snapshot 1`] = `ShallowWrapper {}`;
